### ISSUE TITLE
Python pybel openbabel

### ISSUE THREAD

**Issue by kkempfer** [openbabel/openbabel#2426]
Dear all,

- [x] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: Open Babel 3.0.0 -- Mar 11 2020 -- 11:52:36
Operating system and version: Windows Subsystem for Linux (WSL 2), Ubuntu 20.04 LTS
Python version: Python 3.8.10 (default, Sep 28 2021, 16:10:42)

## Expected Behavior
I want to use 4 methods to draw PNG and SVG images from SMILES. 
- `obabel` command line program
- `obgui` GUI interface
- `openbabel` Python package
- `pybel` Python package

## Actual Behavior
All 4 methods work for SVG files but the two Python-based approaches fail for the PNG file conversion. `openbabel` runs through the end, but the output PNG file is corrupted. Using `pybel` leads to a Segmentation Fault.

## Steps to Reproduce

This works:

    $ obabel -:"COC1=C(C=CC(=C1)C=O)O" -O vanillin.png

The next one runs until the end but the output file is corrupted.

    $ python
    (openbabel) kkempfer@LAPTOP-PPP58QBA:(0)$ python
    Python 3.8.10 (default, Sep 28 2021, 16:10:42)
    [GCC 9.3.0] on linux
    Type "help", "copyright", "credits" or "license" for more information.
    >>> from openbabel import openbabel as ob
    >>> mol = ob.OBMol()
    >>> obConversion = ob.OBConversion()
    >>> obConversion.SetInAndOutFormats("smi", "png")
    True
    >>> obConversion.ReadString(mol, "COC1=C(C=CC(=C1)C=O)O")
    True
    >>> obConversion.WriteFile(mol, "vanillin-ob.png")
    True
    >>> 

`pybel` implementation leads to a segmentation fault:

    $ python
    Python 3.8.10 (default, Sep 28 2021, 16:10:42)
    [GCC 9.3.0] on linux
    Type "help", "copyright", "credits" or "license" for more information.
    >>> from openbabel import pybel
    >>> mol = pybel.readstring("smi", "COC1=C(C=CC(=C1)C=O)O")
    >>> mol.write("png", "vanillin-pybel.png")
    Segmentation fault

**Comment by fredrikw** [comment]
I cannot find it in the documentation right now, but due to the fact that `obabel -opng` commands can both add chemical data to an existing png file and create a png depiction of a molecule there are two "png-formats".

If you modify your code to use `obConversion.SetInAndOutFormats("smi", "_png2")` instead it should work.

**Comment by nbehrnd** [comment]
E.g. `obabel -:"c1ccncn1" -O test.png --append MW`

**Comment by fredrikw** [comment]
That was not the feature I was thinking about. That is just to create a png that normally show the title of the compound but you add MW to the title.
More complex are the following two examples from `obabel -Lpng`
```
Molecules can also be embedded in an existing PNG file::

  obabel existing.png mymol1.smi mymol2.mol -O augmented.png -xO mol

Reading from a PNG file will extract any embedded chemical structure data::

  obabel augmented.png -O contents.sdf
```

**Comment by fredrikw** [comment]
@nbehrnd That looks lika bug...

**Comment by kkempfer** [comment]
> I cannot find it in the documentation right now, but due to the fact that `obabel -opng` commands can both add chemical data to an existing png file and create a png depiction of a molecule there are two "png-formats".
> 
> If you modify your code to use `obConversion.SetInAndOutFormats("smi", "_png2")` instead it should work.

Yes, I confirm that using the _png2 file format works for both pybel and openbabel Python modules. It feels a bit a bit hacky to use a "behind the scenes" format.
The _png2 format is indeed not displayed in the latest documentation: https://open-babel.readthedocs.io/en/latest/FileFormats/Image_Formats.html
But we can see it in this previous release: http://openbabel.org/docs/current/FileFormats/PNG2_format.html

**Comment by nbehrnd** [comment]
@kkempfer The inclusion of a Molfile into a `.png` is mentioned in section 18.8.2 / Image formats - png.  On-line on readthedocs, it is [here](https://open-babel.readthedocs.io/en/latest/FileFormats/PNG_2D_depiction.html); in case you use the `.pdf` manual about release 3.0.1 (June 14, 2020), it is on page 192.
